engine/link/link.asm

```diff
@@ -2127,30 +2133,290 @@ Special_TryQuickSave:
 	ld [wChosenCableClubRoom], a
 	ret
 
-Special_CheckBothSelectedSameRoom:
-	ld a, [wChosenCableClubRoom]
-	call Link_EnsureSync
-	push af
-	call LinkDataReceived
+PerformLinkChecks:
+	xor a
+	ld bc, 10
+	ld hl, wLinkReceivedPolishedMiscBuffer
+	call ByteFill
+
+	; This acts as the old Special_CheckBothSelectedSameRoom
+	; We send a dummy byte here that will cause old versions
+	; of PolishedCrystal's CheckBothSelectedSameRoom function
+	; to fail.
+	ld a, $3
```

Comment:
This could be a defined constant too, with a comment that old polishedcrystal would not have 3 as a possible room selection value.